pkg/catalog/generator.go

```diff
@@ -77,11 +78,35 @@ type SourceCoder interface {
 
 // RootPath appends CurrentDir to output
 func (p *Generator) SourcePath(a SourceCoder) string {
-	rootDir := rootDirectory(path.Join(p.OutputDir, p.CurrentDir))
 	if source_path := Attribute(a, "source_path"); source_path != "" {
-		return rootDir + Attribute(a, "source_path")
+		return rootDirectory(path.Join(p.OutputDir, p.CurrentDir)) + Attribute(a, "source_path")
 	}
-	return path.Join(rootDir, a.GetSourceContext().File)
+
+	return handleSourceURL(a.GetSourceContext().File)
+}
+
+func handleSourceURL(importPath string) string {
+	//FIXME: does not work with external sysl import modules
+	//FIXME: does not work with absolute import in sysl
+	//FIXME: only handles github
+	if strings.HasPrefix(importPath, "github") {
+		urlPath, err := url.Parse("https://" + importPath)
+		if err == nil {
```

Comment:
Nitpick: Could be cleaner with an early return here
```
if err != nil {
    return path.Join(BuildGithubBlobURL(GetRemoteFromGit()), importPath)
}
```

Reply:
also to avoid duplication